src/Analyzers/CSharp/CodeFixes/RemoveUnreachableCode/CSharpRemoveUnreachableCodeCodeFixProvider.cs

```diff
@@ -75,22 +76,36 @@ private static bool IsSubsequentSection(Diagnostic diagnostic)
         {
             foreach (var diagnostic in diagnostics)
             {
-                var firstUnreachableStatementLocation = diagnostic.AdditionalLocations.Single();
+                var firstUnreachableStatementLocation = diagnostic.AdditionalLocations.First();
                 var firstUnreachableStatement = (StatementSyntax)firstUnreachableStatementLocation.FindNode(cancellationToken);
 
-                editor.RemoveNode(firstUnreachableStatement, SyntaxRemoveOptions.KeepUnbalancedDirectives);
+                RemoveStatement(editor, firstUnreachableStatement);
 
                 var sections = RemoveUnreachableCodeHelpers.GetSubsequentUnreachableSections(firstUnreachableStatement);
                 foreach (var section in sections)
                 {
                     foreach (var statement in section)
                     {
-                        editor.RemoveNode(statement, SyntaxRemoveOptions.KeepUnbalancedDirectives);
+                        RemoveStatement(editor, statement);
                     }
                 }
             }
 
             return Task.CompletedTask;
+
+            // Local function
+            static void RemoveStatement(SyntaxEditor editor, SyntaxNode statement)
+            {
+                if (!statement.IsParentKind(SyntaxKind.Block)
+                    && !statement.IsParentKind(SyntaxKind.SwitchSection))
+                {
+                    editor.ReplaceNode(statement, SyntaxFactory.Block());
+                }
+                else
+                {
+                    editor.RemoveNode(statement, SyntaxRemoveOptions.KeepUnbalancedDirectives);
+                }
+            }
```

Comment:
why did this change?

Reply:
Previously we faded this scenario but didn't offer a code fix. In the new implementation it was easier to implement this missing feature than to preserve the exclusion. One of the tests has been updated to show this new behavior.

Reply:
ah ok.